homeassistant/components/media_player/emby.py

```diff
@@ -0,0 +1,370 @@
+
+REQUIREMENTS = []
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+HASS_ID = '055cfe9f2e0413d479e5111d96e7ff6e'
+
+DEFAULT_PORT = 8096
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_EMBY = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_STOP | SUPPORT_SEEK
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup the Emby platform."""
+    host = config.get(CONF_HOST)
+    key = config.get(CONF_API_KEY)
+    port = config.get(CONF_PORT)
+
+    # Build URL
+    url = 'http://{}:{}'.format(host, port)
+
+    _LOGGER.info('Setting up Emby server at: ' + url)
+
+    embyserver = EmbyRemote(key, url)
+
+    # _LOGGER.info('Connected to: %s', host)
+
+    emby_clients = {}
+    emby_sessions = {}
+    track_utc_time_change(hass, lambda now: update_devices(), second=30)
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update_devices():
+        """Update the devices objects."""
+        devices = embyserver.get_sessions()
+        if devices is None:
+            _LOGGER.error('Error listing Emby devices.')
+            return
+
+        new_emby_clients = []
+        for device in devices:
+            if device['DeviceId'] != HASS_ID:
+                if device['DeviceId'] not in emby_clients:
+                    _LOGGER.info('New Emby DeviceID: %s. Adding to Clients.',
+                                 device['DeviceId'])
+                    new_client = EmbyClient(embyserver, device, emby_sessions,
+                                            update_devices, update_sessions)
+                    emby_clients[device['DeviceId']] = new_client
+                    new_emby_clients.append(new_client)
+                else:
+                    emby_clients[device['DeviceId']].set_device(device)
+
+        if new_emby_clients:
+            add_devices_callback(new_emby_clients)
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update_sessions():
+        """Update the sessions objects."""
+        sessions = embyserver.get_sessions()
+        if sessions is None:
+            _LOGGER.error('Error listing Emby sessions')
+            return
+
+        emby_sessions.clear()
+        for session in sessions:
+            emby_sessions[session['DeviceId']] = session
+
+    update_devices()
+    update_sessions()
+
+
+class EmbyRemote:
+    """Emby API Connection Handler."""
+
+    def __init__(self, api_key, server_url):
+        """Initialize Emby API class."""
+        self.api_key = api_key
+        self.server_url = server_url
+
+    @property
+    def get_sessions_url(self):
+        """Return the session url."""
+        return self.server_url + '/Sessions?api_key={0}'
```

Comment:
The parameters are handled as dict. This would mean that you would use `requests.Request().prepare()` to set up the base for the requests and tweak them as needed.